src/generate/TypesGenerator.ts

```diff
@@ -69,8 +69,32 @@ class TypesGenerator {
       throw new Error('Unable to find a directory containing Open API spec files in provided repo');
     }
 
+    return await this.getSpecFileNames('reference', targetDir);
+  }
+
+  /**
+   * Returns a list of valid Open API Spec files URLs that could exist within directories or as standalone files.
+   *
+   * @returns Array of Open API Spec file URLs as strings
+   */
+
+  private async getSpecFileNames(targetDirPath: string, targetDir: NestedTree): Promise<string[]> {
     const remoteSourceDir = await this.client.get<GitHubRepoTree>(targetDir.url);
-    const specFileNames = remoteSourceDir.data.tree.map(file => `${this.CONTENT_URL}${file.path}`);
+
+    const specFileNames = await remoteSourceDir.data.tree.reduce(
```

Comment:
🍹 since this is a deeply nested prop, should we consider adding in optional chaining?

Reply:
I think this should be fine, based on the GIT trees api, the elements that we are chaining should be defined as either a tree type or string type👍🏼 . Reference: https://docs.github.com/en/rest/git/trees?apiVersion=2022-11-28